### PR TITLE
{tools}[foss/2021a] networkx v2.5.1 w/ Python 3.9.5

### PR DESCRIPTION
**Pull request by robert-mijakovic**
(created using `eb --new-pr`)

### CODE REVIEW

**Comment by robert-mijakovic** [comment]
@boegelbot please test @ generoso

File: easybuild/easyconfigs/n/networkx/networkx-2.5.1_pin_decorator.patch [hunk 1]
```diff
@@ -0,0 +1,19 @@
+From caffb25cae8c6fa783db1489136ae41bbd17fd23 Mon Sep 17 00:00:00 2001
+From: Ross Barnowski <rossbar@berkeley.edu>
+Date: Tue, 18 May 2021 10:52:23 -0700
+Subject: [PATCH] Pin upper bound on decorator for 2.6 release.
+
+---
+ requirements/default.txt | 2 +-
+ 1 file changed, 1 insertion(+), 1 deletion(-)
+
+diff --git a/requirements/default.txt b/requirements/default.txt
```
Comment (Micket) [inline easybuild/easyconfigs/n/networkx/networkx-2.5.1_pin_decorator.patch:10]: this file doesn't seem to exist?

**Review by Micket — CHANGES_REQUESTED** [review]

File: easybuild/easyconfigs/n/networkx/networkx-2.5.1_pin_decorator.patch (hunk 1 above)
Comment (verdurin) [reply easybuild/easyconfigs/n/networkx/networkx-2.5.1_pin_decorator.patch:10]: @robert-mijakovic any update on this?

Comment (robert-mijakovic) [reply easybuild/easyconfigs/n/networkx/networkx-2.5.1_pin_decorator.patch]: Solved!

**Comment by robert-mijakovic** [comment]
@boegelbot please test @ generoso

**Comment by robert-mijakovic** [comment]
@boegelbot please test @ generoso

**Comment by robert-mijakovic** [comment]
@Micket I have done the changes you requested.

**Comment by verdurin** [comment]
Test report by @verdurin
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
nuc.lan - Linux Fedora 33, x86_64, Intel(R) Core(TM) i7-8650U CPU @ 1.90GHz, Python 3.9.5
See https://gist.github.com/1a6c55f7a524314d84e78af1d8f3e6a4 for a full test report.

File: easybuild/easyconfigs/n/networkx/networkx-2.5.1_decorator.patch [hunk 2]
```diff
@@ -0,0 +1,21 @@
+diff -Naur networkx-2.5.1.orig/networkx.egg-info/requires.txt networkx-2.5.1/networkx.egg-info/requires.txt
```
Comment (verdurin) [inline easybuild/easyconfigs/n/networkx/networkx-2.5.1_decorator.patch:1]: @robert-mijakovic could you add a comment about the purpose of the patch please?

**Review by verdurin — DISMISSED** [review]

File: easybuild/easyconfigs/n/networkx/networkx-2.5.1_decorator.patch (hunk 2 above)
Comment (robert-mijakovic) [reply easybuild/easyconfigs/n/networkx/networkx-2.5.1_decorator.patch:1]: I don't think networkx 2.5.1 is relevant any longer. networkx 2.6.1 is released and it doesn't require a patch. I will update the pull request.

**Comment by smoors** [comment]
@boegelbot: please test @ generoso

**Comment by hajgato** [comment]
@smoors Not sure that it is important but `networkx` `2.6` and `2.6.1` was yanked.

**Comment by hajgato** [comment]
see https://github.com/networkx/networkx/pull/4967

**Comment by hajgato** [comment]
@robert-mijakovic ^& ^^

**Comment by smoors** [comment]
@hajgato @robert-mijakovic  thanks for the heads up.

I think the circular dependency issue is not important for us because in EB Python packages are installed with `--no-deps`, which avoids it. this version is indeed marked as yanked but the sources are still available (as it should be).

on the other hand, it would be good to get rid of the NumPy, SciPy, Matplotlib, and Pandas deps, so we can move this to `GCCcore` level (or even add to Python itself?) since so many packages depend on it. so it's probably a good idea to wait for version 2.6.2 to come out.

**Comment by robert-mijakovic** [comment]
Ok, then let's wait for networkx 2.6.2.

**Comment by hajgato** [comment]
Or just use `2.5.1` ? God knows when `2.6.2` would come out. This PR sits here for almost a month, and it is a blocker for me.

**Comment by robert-mijakovic** [comment]
@smoors @verdurin @hajgato What do you think? Should we go step back and take 2.5.1?

**Comment by smoors** [comment]
`2.5.1` is also fine for me.

**Comment by robert-mijakovic** [comment]
@smoors Ok, done! Let's merge it not to block people.

**Comment by smoors** [comment]
@boegelbot: please test @ generoso

**Comment by smoors** [comment]
Test report by @smoors
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
node381.hydra.os - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) Gold 6148 CPU @ 2.40GHz (skylake_avx512), Python 2.7.5
See https://gist.github.com/44aa6a42ef165183eb52938d081251ba for a full test report.

**Review by smoors — APPROVED** [review]
lgtm

**Comment by smoors** [comment]
Going in, thanks @robert-mijakovic!

**Pull request merged**